feat: Add infos about dnf-plugins-core (#1156)
rocky-linux/documentation: docs/books/admin_guide/13-softwares.md

### Conclusion

EPEL is not an official repository for RHEL. But it can be useful for administrators and developers who work with RHEL or derivatives and need some utilities prepared for RHEL from a source they can feel confident about.

## DNF Plugins

The `dnf-plugins-core` package adds plugins to `dnf` that will be useful for managing your repositories.

> !!! NOTE

Author: ```suggestion
!!! NOTE

```

    See more informations here: https://dnf-plugins-core.readthedocs.io/en/latest/index.html

> 

Reviewer: Hey @alemorvan, for the most part this looks good. I didn't have a chance to test anything and no time to really look hard at it. I *think* that you and I decided that for admonitions we needed the content to be one line down from the admonition type like this:
```
!!! Note
    
    See more information here ...
```
If I recall, this had something to do with how Crowdin messed with admonitions.

Author: I did the change.
Thx

### `config-manager` plugin

Manage DNF options, add repos, or disable them.

Examples:

* Download a `.repo` file and use it:

```
dnf config-manager --add-repo https://packages._centreon_.com/rpm-standard/23.04/el8/_centreon_.repo
```

* You can also set an url as a base url for a repo:

```
dnf config-manager --add-repo https://repo.rocky.lan/repo
```

* Enable or disable one or more repos:

```
dnf config-manager --set-enabled epel centreon
dnf config-manager --set-disabled epel centreon
```

* Add a proxy to your config file:

```
dnf config-manager --save --setopt=*.proxy=http://proxy.rocky.lan:3128/
```

### `copr` plugin

`copr` is an automatic rpm forge, providing a repo with built packages.

* Activate a copr repo:

```
copr enable xxxx
```

### `download` plugin

Download rpm package instead of installing it:

```
dnf download ansible
```

If you just want to obtain the remote location url of the package:

```
dnf download --url ansible
```

Or if you want to also download the dependencies:

```
dnf download --resolv --alldeps ansible
```

### `needs-restarting` plugin

> After running an `dnf update`, the running processes will continue to run but with the old binaries. In order to take into account the code changes and especially the security updates, they have to be restarted.

Reviewer 2: ```suggestion
After running a `dnf update`, the running processes will continue to run but with the old binaries. In order to take into account the code changes and especially the security updates, they have to be restarted.
```

Reviewer 3: I'd add for after a `dnf update` (or any package update really) :
`rpmconf -a -c` (not sure if this pkg is default or from EPEL)
`dnf needs-restarting -s` to check if services need restarting
`dnf needs-restarting -r` to check if a reboot may be required
`dnf needs-restarting -s -r` to do both in one run

Author: @NezSez:
I want to keep as simple as possible this admin guide. I think your use case for rpmconf -a -c (never needed to use it personnaly) is maybe something that should not happened? In that case, a guide would be better.

Author: @NezSez your descriptions for needs-restarting are better than mine.
Please start a review in the github UI and make a suggestion as Serge did just before you.
Like that, the changes will be commited and will belong to you in a git-blame

Reviewer 3: Yes, I'm familiar with git, not so much GitHub, and I didn't have time to figure out it's interface, so I just made the comment.  I also haven't read this admin book so not familiar with context and figured you or Serge would know what to do with the info.

The `needs-restarting` plugin will allow you to detect processes that are in this case.

```
dnf needs-restarting [-u] [-r] [-s]
```

| Options | Description                                                                  |
| --------| --------------------------------------------------------------------------- |
| `-u`     | Only consider processes belonging to the running user.                       |

> | `-r`     | Only report whether a reboot is required (exit code 1) or not (exit code 0). |

Author: ```suggestion
| `-r`     |  to check if a reboot may be required |
```

> | `-s`     | Only list the affected systemd services.                                     |

Author: ```suggestion
| `-s`     | to check if services need restarting                                  |
| `-s -r` | to do both in one run                                                       |
```

### `versionlock` plugin

Sometimes it is useful to protect packages from all updates or to exclude certain versions of a package (because of known problems for example). For this purpose, the versionlock plugin will be of great help. 

You need to install an extra package:

```
python3-dnf-plugin-versionlock.noarch
```

> 

Reviewer: Line 1003... is that correct? (again, no time to test it out at the moment) or should it be `dnf install python3-dnf-plugin-versionlock`?

Examples:

* Lock the ansible version:

```
dnf versionlock add ansible
Adding versionlock on: ansible-0:6.3.0-2.el9.*
```

* List locked packages:

```
dnf versionlock list
ansible-0:6.3.0-2.el9.*

> ```

Reviewer: Overall, I don't see problems with language issues. Perhaps @SergeCroise or @Grammaresque can look it over from a different perspective. Thanks for your work!!

```

### `needs-restarting` plugin

After running a `dnf update`, the running processes will continue to run but with the old binaries. In order to take into account the code changes and especially the security updates, they have to be restarted.

The `needs-restarting` plugin will allow you to detect processes that are in this case.

```
dnf needs-restarting [-u] [-r] [-s]
```

| Options | Description                                                                  |
| --------| --------------------------------------------------------------------------- |
| `-u`     | Only consider processes belonging to the running user.                       |
| `-r`     |  to check if a reboot may be required. |
| `-s`     | to check if services need restarting.                                  |
| `-s -r` | to do both in one run.                                                       |

### `versionlock` plugin

Sometimes it is useful to protect packages from all updates or to exclude certain versions of a package (because of known problems for example). For this purpose, the versionlock plugin will be of great help. 

You need to install an extra package:

```

> python3-dnf-plugin-versionlock.noarch

Author: ```suggestion
dnf install python3-dnf-plugin-versionlock
```
